### ISSUE TITLE
FacetQuery generates syntax error in Solr request parameters

### ISSUE THREAD

**Issue by inghamn** [solariumphp/solarium#936]
**Solarium version(s) affected**: 6.1.2
**Solr version**:7.4
**PHP version**: 7.4

**Description**  
I'm adapting my solr code to use Solarium. It seems like the facetQuery api generates invalid solr syntax for the request URI parameters.

I am able to do the facet query in the Solr interface, but when I attempt to use solarium, I get a syntax error from my Solr server.  Granted it's a complex query, but I use it successfully in my previous code and in the Solr web interface.

**How to reproduce**  
The facet query is:
{!frange l=0}if(exists(slaDays),sub(if(exists(closedDate),ms(closedDate,enteredDate),ms(NOW,enteredDate)),product(slaDays,86400000)),-1)

When I do this in the Solr web interface, or my previous PHP code, it succeeds with Solr recording the log message:
```
2021-06-25 19:46:11.184 INFO  (qtp136157810-20) [   x:ureport] o.a.s.c.S.Request [ureport]  webapp=/solr path=/select params={facet.query={!frange+l%3D0}if(exists(slaDays),sub(if(exists(closedDate),ms(closedDate,enteredDate),ms(NOW,enteredDate)),product(slaDays,86400000)),-1)&q=*:*&facet=on&_=1624650327515} hits=123192 status=0 QTime=45
```

When I do this with Solarium I get a syntax error in the request parameters sent to Solr:
```php
<?php
declare (strict_types=1);
include './bootstrap.inc';

$solr = new \Solarium\Client(
    new \Solarium\Core\Client\Adapter\Curl(),
    new \Symfony\Component\EventDispatcher\EventDispatcher(),
    [
        'endpoint' => [
            'solr' => [
                'host' => SOLR_SERVER_HOST,
                'port' => SOLR_SERVER_PORT,
                'path' => '/',
                'core' => SOLR_SERVER_CORE
            ]
        ]
    ]
);

$query  = '{!frange l=0}if(exists(slaDays),sub(if(exists(closedDate),ms(closedDate,enteredDate),ms(NOW,enteredDate)),product(slaDays,86400000)),-1)';

$select = $solr->createSelect();
$facets = $select->getFacetSet();
$facets->createFacetQuery('overdue')->setQuery($query);
$result = $solr->select($select);

print_r($result->getNumFound())."\n";
print_r($result->getFacetSet()->getFacets());
```
```
2021-06-25 19:33:54.032 INFO  (qtp136157810-48) [   x:ureport] o.a.s.c.S.Request [ureport]  webapp=/solr path=/select params={facet.query={!key%3Doverdue}{!frange+l%3D0}if(exists(slaDays),sub(if(exists(closedDate),ms(closedDate,enteredDate),ms(NOW,enteredDate)),product(slaDays,86400000)),-1)&q=*:*&json.nl=flat&omitHeader=true&fl=*,score&start=0&rows=10&wt=json&facet=true} hits=123192 status=400 QTime=6
```

```
2021-06-25 19:33:54.032 INFO  (qtp136157810-48) [   x:ureport] o.a.s.c.S.Request [ureport]  webapp=/solr path=/select params={facet.query={!key%3Doverdue}{!frange+l%3D0}if(exists(slaDays),sub(if(exists(closedDate),ms(closedDate,enteredDate),ms(NOW,enteredDate)),product(slaDays,86400000)),-1)&q=*:*&json.nl=flat&omitHeader=true&fl=*,score&start=0&rows=10&wt=json&facet=true} hits=123192 status=400 QTime=6
2021-06-25 19:38:56.292 ERROR (qtp136157810-19) [   x:ureport] o.a.s.h.RequestHandlerBase org.apache.solr.common.SolrException: org.apache.solr.search.SyntaxError: Expected ')' at position 17 in 'if(exists(slaDays'
        at org.apache.solr.handler.component.FacetComponent.getFacetCounts(FacetComponent.java:337)
        at org.apache.solr.handler.component.FacetComponent.process(FacetComponent.java:273)
        at org.apache.solr.handler.component.SearchHandler.handleRequestBody(SearchHandler.java:298)
        at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:199)
        at org.apache.solr.core.SolrCore.execute(SolrCore.java:2539)
        at org.apache.solr.servlet.HttpSolrCall.execute(HttpSolrCall.java:709)
        at org.apache.solr.servlet.HttpSolrCall.call(HttpSolrCall.java:515)
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:377)
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:323)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:219)
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:335)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.Server.handle(Server.java:531)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:352)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:760)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:678)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.solr.search.SyntaxError: Expected ')' at position 17 in 'if(exists(slaDays'
        at org.apache.solr.search.StrParser.expect(StrParser.java:78)
        at org.apache.solr.search.FunctionQParser.parseValueSource(FunctionQParser.java:371)
        at org.apache.solr.search.FunctionQParser.parseValueSource(FunctionQParser.java:237)
        at org.apache.solr.search.ValueSourceParser$74.parse(ValueSourceParser.java:873)
        at org.apache.solr.search.FunctionQParser.parseValueSource(FunctionQParser.java:370)
        at org.apache.solr.search.FunctionQParser.parse(FunctionQParser.java:82)
        at org.apache.solr.search.QParser.getQuery(QParser.java:169)
        at org.apache.solr.search.FunctionRangeQParserPlugin$1.parse(FunctionRangeQParserPlugin.java:51)
        at org.apache.solr.search.QParser.getQuery(QParser.java:169)
        at org.apache.solr.parser.SolrQueryParserBase.getLocalParams(SolrQueryParserBase.java:1249)
        at org.apache.solr.parser.QueryParser.Clause(QueryParser.java:321)
        at org.apache.solr.parser.QueryParser.Query(QueryParser.java:162)
        at org.apache.solr.parser.QueryParser.TopLevelQuery(QueryParser.java:131)
        at org.apache.solr.parser.SolrQueryParserBase.parse(SolrQueryParserBase.java:254)
        at org.apache.solr.search.LuceneQParser.parse(LuceneQParser.java:49)
        at org.apache.solr.search.QParser.getQuery(QParser.java:169)
        at org.apache.solr.request.SimpleFacets.getFacetQueryCount(SimpleFacets.java:312)
        at org.apache.solr.request.SimpleFacets.getFacetQueryCounts(SimpleFacets.java:302)
        at org.apache.solr.handler.component.FacetComponent.getFacetCounts(FacetComponent.java:317)
        ... 37 more
```

**Comment by thomascorthals** [comment]
This is the same problem reported in #229. You'll find a hack there that might help you out.

**Comment by inghamn** [comment]
Thanks so much, @mkalkbrenner for the fix.

**Comment by thomascorthals** [comment]
Fixed in [Solarium 6.1.4](https://github.com/solariumphp/solarium/releases/tag/6.1.4).
